modules/exploits/multi/http/spip_bigup_unauth_rce.rb

```diff
@@ -0,0 +1,179 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Payload::Php
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HTTP::Spip
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'SPIP BigUp Plugin Unauthenticated RCE',
+        'Description' => %q{
+          This module exploits a Remote Code Execution vulnerability in the BigUp plugin of SPIP.
+          The vulnerability lies in the `lister_fichiers_par_champs` function, which is triggered
+          when the `bigup_retrouver_fichiers` parameter is set to `1`. By exploiting the improper
+          handling of multipart form data in file uploads, an attacker can inject and execute
+          arbitrary PHP code on the target server.
+
+          This critical vulnerability affects all versions of SPIP from 4.0 up to and including
+          4.3.1, 4.2.15, and 4.1.17. It allows unauthenticated users to execute arbitrary code
+          remotely via the public interface. The vulnerability has been patched in versions
+          4.3.2, 4.2.16, and 4.1.18.
+        },
+        'Author' => [
+          'Vozec', # Vulnerability Discoverer
+          'Laluka', # Vulnerability Discoverer
+          'Julien Voisin', # Code Review
+          'Valentin Lobstein' # Metasploit Module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['URL', 'https://blog.spip.net/Mise-a-jour-critique-de-securite-sortie-de-SPIP-4-3-2-SPIP-4-2-16-SPIP-4-1-18.html']
```

Comment:
Should add:

* https://thinkloveshare.com/hacking/spip_preauth_rce_2024_part_2_a_big_upload/
* CVE-2024-8517

Reply:
Added in https://github.com/rapid7/metasploit-framework/pull/19444/commits/92a25e5a296165006f7141b2db51c1528e5f6f4d